Support for setting color

Hi, it would be nice to have a support for changing a color. I am utilizing RGBW LED strips, which I can't control ATM from my Home Assistant instance which I am sad about.

BTW, moc díky za knihovnu i za plugin pro HA! :)

Hi, @tomasbedrich, it took me a while to find a bit of free time and mental energy after work to look at this.

So I clonned this repo, adjusted IP of relay. Not sure how can I find number of zones and it looks like it is obtained automatically, isn't it?
Next, running `make test` I am getting a "poetry" error:
```
$ make test
make: poetry: Command not found
Makefile:40: recipe for target 'venv/.poetry-4fa4c8d0cc72f208bef0793e07c685f114e66c01' failed
make: *** [venv/.poetry-4fa4c8d0cc72f208bef0793e07c685f114e66c01] Error 127
```
I am running on Ubuntu-based distro and I use virtual python environment with dependencies for *skydance* installed there.

Any idea what's wrong here?

--- a/docs/api/protocol.md
+++ b/docs/api/protocol.md
@@ -25,10 +25,10 @@
 ::: skydance.protocol.TemperatureCommand
 ::: skydance.protocol.RGBWCommand
 ::: skydance.protocol.GetNumberOfZonesCommand
-::: skydance.protocol.GetZoneNameCommand
+::: skydance.protocol.GetZoneInfoCommand
 
 ## Responses
 
 ::: skydance.protocol.Response
 ::: skydance.protocol.GetNumberOfZonesResponse
-::: skydance.protocol.GetZoneNameResponse
+::: skydance.protocol.GetZoneInfoResponse

--- a/skydance/enum.py
+++ b/skydance/enum.py
@@ -1,0 +1,10 @@
+from enum import Enum
+
+
+class ZoneType(Enum):
+    Switch = 0x01
+    Dimmer = 0x11
+    CCT = 0x21
+    RGB = 0x31
+    RGBW = 0x41
+    RGBCCT = 0x51

--- a/skydance/protocol.py
+++ b/skydance/protocol.py
@@ -2,6 +2,8 @@ import struct
 from abc import ABCMeta, abstractmethod
 from functools import partial
 
+from skydance.enum import ZoneType
+
 
 PORT = 8899
 """A port used for communication with a relay."""
@@ -333,7 +335,7 @@ class GetNumberOfZonesCommand(Command):
         )
 
 
-class GetZoneNameCommand(ZoneCommand):
+class GetZoneInfoCommand(ZoneCommand):
     """Discover a zone according to it's number."""
 
     @property
@@ -412,16 +414,20 @@ class GetNumberOfZonesResponse(Response):
         return sum(1 for zone in self.body[12:28] if zone != 0)
 
 
-class GetZoneNameResponse(Response):
+class GetZoneInfoResponse(Response):
     """
-    Parse a response for `GetZoneNameCommand`.
+    Parse a response for `GetZoneInfoCommand`.
 
-    See: [`GetZoneNameCommand`][skydance.protocol.GetZoneNameCommand].
+    See: [`GetZoneInfoCommand`][skydance.protocol.GetZoneInfoCommand].
     """
 
-    # Name offset experimentally decoded from response packets.
+    @property
+    def type(self) -> ZoneType:
+        """Return a zone type."""
+        return ZoneType(self.body[12])
 
     @property
     def name(self) -> str:
         """Return a zone name."""
+        # Name offset experimentally decoded from response packets.
         return self.body[14:].decode("utf-8", errors="replace").strip(" \x00")

--- a/skydance/tests/test_protocol.py
+++ b/skydance/tests/test_protocol.py
@@ -179,7 +179,7 @@ def test_get_number_of_zones_response(response, num):
     range(1, 17),
 )
 def test_get_zone_name(state, zone: int):
-    res = GetZoneNameCommand(state, zone=zone).body
+    res = GetZoneInfoCommand(state, zone=zone).body
     zone_encoded = 2 ** (zone - 1)
     expected = bytes().join(
         (
@@ -197,7 +197,7 @@ def test_get_zone_name(state, zone: int):
 )
 def test_get_zone_name_invalid(zone: int):
     with pytest.raises(expected_exception=ValueError):
-        GetZoneNameCommand.validate_zone(zone)
+        GetZoneInfoCommand.validate_zone(zone)
 
 
 @pytest.mark.parametrize(
@@ -214,12 +214,33 @@ def test_get_zone_name_invalid(zone: int):
         ),
     ],
 )
-def test_get_zone_name_response_strip(variant):
-    assert GetZoneNameResponse(variant).name == "Zone RGB+CCT"
+def test_get_zone_info_response_strip(variant):
+    zone_info = GetZoneInfoResponse(variant)
+    assert zone_info.name == "Zone RGB+CCT"
 
 
-def test_get_zone_name_response_utf_8():
+def test_get_zone_info_response_utf_8():
     raw = bytes.fromhex(
         "55aa5aa57e02800080e18026510200f8100021004b75636879c58820746f70000000007e"
     )
-    assert GetZoneNameResponse(raw).name == "Kuchyň top"
+    zone_info = GetZoneInfoResponse(raw)
+    assert zone_info.name == "Kuchyň top"
+
+
+zone_info_params = {
+    "55aa5aa57e568000805dc126518000f8100001005a6f6e6520537769746368000000007e": ZoneType.Switch,
+    "55aa5aa57e05800080e18026511000f8100011005a6f6e652044696d6d6572000000007e": ZoneType.Dimmer,
+    "55aa5aa57e528000805dc126510800f8100021005a6f6e6520434354000000000000007e": ZoneType.CCT,
+    "55aa5aa57e538000805dc126511000f8100031005a6f6e6520524742000000000000007e": ZoneType.RGB,
+    "55aa5aa57e548000805dc126512000f8100041005a6f6e6520524742570000000000007e": ZoneType.RGBW,
+    "55aa5aa57e00800080e18026514000f8100051005a6f6e65205247422b4343540000007e": ZoneType.RGBCCT,
+}
+
+
+@pytest.mark.parametrize(
+    "raw, expected_type",
+    [(bytes.fromhex(raw), zt) for raw, zt in zone_info_params.items()],
+)
+def test_get_zone_info_response_types(raw, expected_type):
+    zone_info = GetZoneInfoResponse(raw)
+    assert zone_info.type == expected_type

--- a/tests/test_manual.py
+++ b/tests/test_manual.py
@@ -77,13 +77,13 @@ async def test_zone_discovery(state, session):
     number_of_zones = GetNumberOfZonesResponse(res).number
 
     for zone in range(1, number_of_zones + 1):
-        log.info("Getting name of zone=%d", zone)
-        cmd = GetZoneNameCommand(state, zone=zone).raw
+        log.info("Getting info about zone=%d", zone)
+        cmd = GetZoneInfoCommand(state, zone=zone).raw
         await session.write(cmd)
         state.increment_frame_number()
         res = await session.read(64)
-        zone_name = GetZoneNameResponse(res).name
-        log.info("Zone=%d has name=%s", zone, zone_name)
+        zone_info = GetZoneInfoResponse(res)
+        log.info("Zone=%d has type=%s, name=%s", zone, zone_info.type, zone_info.name)
 
 
 @pytest.mark.asyncio

